festim/hydrogen_transport_problem.py

```diff
@@ -257,13 +279,17 @@ def run(self, final_time: float):
         progress = tqdm.autonotebook.tqdm(
             desc="Solving H transport problem", total=final_time
         )
-        while t < final_time:
+        while float(self.t) < final_time:
```

Comment:
Its a matter of preference, but seen as t is a fem.Constant, we can access the value by using `self.t.value` as is done in L284, same with dt. Either way we should be consistent